src/core/Akka/Actor/ActorCell.DefaultMessages.cs

```diff
@@ -214,6 +215,9 @@ private void HandleActorTaskSchedulerMessage(ActorTaskSchedulerMessage m)
         {
             if (m.Exception != null)
             {
+                //set the current message captured in the async operation
+                //current message was cleared earlier when the async receive handler completed
+                CurrentMessage = m.Message;
                 HandleInvokeFailure(m.Exception);
```

Comment:
If we are handling a `ActorTaskSchedulerMessage`, we need to re-set the current message captured in the async operation. 
The actorcells current message was cleared when the async receive method completed.